proc/api.thrift

```diff
@@ -0,0 +1,40 @@
+service Proc {
+  
+  exception InvalidRequest {
+    1: optional string message
+  }
+
+  exception ScootTooBusy {
+    1: optional i64 retryAfterMs
+  }
+  
+  enum JobType {
+    IRON_TESTS=1,
+    DEFAULT=2,
+  }
+
+  struct Command {
+    1: list<string> cmds
+  }
+
+  struct Task {
+    1: optional string id
+    2: required Command command,
+    3: optional string snapshotId,
+    4: optional string contextId
+  }
+
+  struct Job {
+    1: required string id,
+    2: required list<Task> tasks
+  }
+
+  Job RunJob(
+    1: optional string contextId,
+    2: optional JobType jobType,
+    3: required list<Task> tasks
```

Comment:
At some points I've heard that required is really hard to ever upgrade, and so a large school of thought is to make everything optional; wdyt?